dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/DatasetService.java

```diff
@@ -112,12 +117,27 @@ public AddFilesResult addFilesToDataset(long datasetId, List<DataFile> newFiles)
                 hasProvenanceErrors = true;
             }
         }
-        
+
         return new AddFilesResult(dataset, notSavedFilesCount, hasProvenanceErrors);
     }
-    
+
+    /**
+     * Replaces thumbnail (default if none is set) with the one provided.
+     *
+     * @param datasetForNewThumbnail dataset that will have new thumbnail
+     * @param datafileThumbnailId    id of the thumbnail that will be set for dataset
+     */
+    public DatasetThumbnail changeDatasetThumbnail(Dataset datasetForNewThumbnail, long datafileThumbnailId) {
```

Comment:
I think I will go with the way I did in last PR, since we already have dataset fetched in the Page since we have to, it would be kinda shame to fetch it again IMO. If we would only use REST Api we would fetch all the entities in the services.